### DISCUSSION TITLE
enableHiding seems to mean enableTogglingVisiblity

### DISCUSSION THREAD

**Discussion by slhsxcmy** [TanStack/table#5365] [Q&A]
I have a use case where I want to filter rows based on fields that are not displayed in columns. I found a working solution by 
- defining these fields as "columns"
- set visibility to false
- set enableHiding to false

I realized that with default visibility false, "enableHiding" really means "enableShowing", so in general it means "enableTogglingVisiblity". Is this intended behavior? Is there a better way to filter fields that are not displayed in columns?

**Comment by KevinVandy** [comment]
This is intended behavior. As far as what's best for filtering the way you want to, that's one ok way to do it. You can also just modify another columns' filterFn to compare values from `row.original`, so that you can filter on whatever criteria is needed.

**Comment by slhsxcmy** [comment]
Thank you for the clarification! I realized that table.getToggleAllColumnsVisibilityHandler() doesn't work, I guess it's because some columns's visibility cannot be toggled. Is that intended?
